src/xform_to_datalog/authorization_logic_datalog_facts_test.cc

```diff
@@ -0,0 +1,55 @@
+//-----------------------------------------------------------------------------
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//-----------------------------------------------------------------------------
+#include "src/xform_to_datalog/authorization_logic_datalog_facts.h"
+
+#include <cstdlib>
+#include <filesystem>
+#include <fstream>
+#include <vector>
+
+#include "absl/strings/str_split.h"
+#include "src/common/testing/gtest.h"
+#include "src/xform_to_datalog/authorization_logic_test.h"
+
+namespace fs = std::filesystem;
+
+namespace raksha::xform_to_datalog {
+
+class AuthorizationLogicDatalogFactsTest : public AuthorizationLogicTest {};
+
+TEST_F(AuthorizationLogicDatalogFactsTest, InvokesRustToolAndGeneratesOutput) {
+  for (const std::string& program : {"simple_auth_logic", "empty_auth_logic"}) {
+    const fs::path& test_data_dir = GetTestDataDir();
```

Comment:
No need to use a reference here. We are returning a temporary value from `GetTestDataDir`, so it is friendlier to copy elision to make this just `fs::path test_data_dir`.

Reply:
Done. However, won't this get RVOed even with `const &` assignment?

Reply:
It's less about the performance here and more about other functions that `test_data_dir` might be used as an argument to. In those cases, having it be a value rather than a reference/pointer gives the compiler more leeway to elide copies at the callsite.